project_template/CMakeLists.txt

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+cmake_minimum_required(VERSION 3.15)
+
+project(az_exampleshortname
+    LANGUAGES C CXX
+    VERSION 0.1.0)
+
+option(az_exampleshortname_with_docs "Build documentation" ON)
+option(BUILD_TESTING "Build tests" ON)
+
+
+
+
+
+# this is a cmake 3.15 feature that tells find_package to 
+# look at config files before module files. This makes it easier
+# to write buildsystems that work on many platforms, espicially
+# when finding libraries that have multiple build systems.
+
+
+# if needed: add ./cmake to the module path
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_LIST_DIR}/cmake)
+
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_CXX_STANDARD 17)
+
+# default to hidden symbols when building shared libraries
+# on unix systems

```

Comment:
Does the below have different behavior on non-unix systems? If not, we should simply strike this line.

Reply:
on windows, there is no way to build with "default visibility" as exported. This makes the unix behavior more consistent with the windows behavior (and more correct)

Reply:
So how about `# symbols in built shared libraries are hidden by default`?